fondant/component.py

```diff
@@ -74,27 +74,37 @@ def _load_or_create_manifest(self) -> Manifest:
         """Abstract method that returns the dataset manifest"""
 
     @abstractmethod
-    def _process_dataset(self, dataset: FondantDataset) -> FondantDataset:
+    def _process_dataset(self, dataset: FondantDataset) -> dd.DataFrame:
         """Abstract method that processes the input dataframe and updates the Fondant Dataset
         with the new or loaded subsets"""
 
     def run(self):
         """
         Runs the component.
         """
-        manifest = self._load_or_create_manifest()
-        dataset = FondantDataset(manifest)
-        dataset = self._process_dataset(dataset)
+        input_manifest = self._load_or_create_manifest()
+        dataset = FondantDataset(input_manifest)
+
+        df = self._process_dataset(dataset)
+
+        output_manifest = input_manifest.evolve(component_spec=self.spec)
+        dataset = FondantDataset(output_manifest)
+
+        # write index and output subsets to remote storage
+        dataset.write_index(df)
+        dataset.write_subsets(df, self.spec)
+
         dataset.upload(save_path=self.args.output_manifest_path)
```

Comment:
I would lift saving the manifest from the `Dataset` to the `Component` class.

Reply:
Done!